File: Java/source/src/main/java/com/yitter/test/GenTest.java
package com.yitter.test;

import com.yitter.contract.IIdGenerator;

public class GenTest {

    private IIdGenerator IdGen;
    private int GenIdCount;
    private int WorkerId;

    public GenTest(IIdGenerator idGen, int genIdCount, int workerId) {
        GenIdCount = genIdCount;
        IdGen = idGen;
        WorkerId = workerId;
    }

    public void GenStart() {
        long start = System.currentTimeMillis();

        for (int i = 0; i < GenIdCount; i++) {
            long id = IdGen.newLong();
        }

        long end = System.currentTimeMillis();
        long time = end - start;

        System.out.println("++++++++++++++++++++++++++++++++++++++++WorkerId: "
                + WorkerId + ", total: " + time + " ms");

    }
}

File: C#/source/Yitter.IdGenTest/GenTest.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using Yitter.IdGenerator;

namespace Yitter.OrgSystem.TestA
{
    public class GenTest
    {
        private IIdGenerator IdGen;
        private Hashtable ids = new Hashtable();
        public IList<long> idList = new List<long>();
        private int GenIdCount;
        private int WorkerId;
            

        public GenTest(IIdGenerator idGen, int genIdCount, int workerId)
        {
            GenIdCount = genIdCount;
            IdGen = idGen;
            WorkerId = workerId;
        }

        //public void GenId()
        //{
        //    Thread t = new Thread(new ThreadStart(Gen1Start));
        //    t.Start();
        //}

        public void GenStart()
        {
            DateTime start = DateTime.Now;

            for (int i = 0; i < GenIdCount; i++)
            {
                var id = IdGen.NewLong();
                //ids.Add(id, i);
                //idList.Add(id);
            }

            DateTime end = DateTime.Now;
            Console.WriteLine($"++++++++++++++++++++++++++++++++++++++++WorkerId: {WorkerId}, total: {(end - start).TotalSeconds} s");
            Interlocked.Increment(ref Program.Count);
        }
    }
}
